### PR TITLE
PDF generation scripts

### PR DESCRIPTION
**Pull request by dcbr**
I think I went down the rabbithole a bit too far with this PR, so sorry for the huge diff. But I think the end result is quite nice and I learned a lot about pdf structure along the way :)

## Summary:

- Backend: A generic script is added to generate the TOB pdfs from scratch. Modifications to the original pdf files are provided for all languages (dutch, french, german and english).
- Form changes: All textfields now have the same height (and font size).
- Form changes: Months in the heading are now formatted with a leading zero.
- Form changes: Extra textfields for table b (taxes with cap) are added, but not used/fillable yet at this moment. That's a first step in dealing with #11, but the logic implementation is definitely for another PR.
- Form changes: Extra strikethrough annotations are added to clarify the identification.
- Form changes: The files are now PDF/A compliant after filling the PDF. The document id is also made unique for each generated pdf and the metadata is updated to reflect the proper time of creation/modification.
- Form changes: The signature name and capacity text fields are merged, such that the capacity appears right next to the name now.
- Form changes & frontend: The `formatMoney` function is changed to support "thousand separators". The `fillPdf` function uses '|' as the thousand separator, to bring the style closer to the original forms.
- Frontend: language selection box to switch between the different TOB forms.

### CODE REVIEW

File: src/formatting.ts [hunk 1]
```diff
@@ -1,12 +1,10 @@
...
-    // Math.ceil(1_99.99 % 100 = 99.99) = 100
-    // Then we want to change the value to 2_00 and make the remainder 0
-    if(decimals == 100) {
-        decimals = 0;
-        value += 1;
-    }
-    const formattedDecimals = decimals < 10 ? `0${decimals}` : `${decimals}`;
+export function formatMoney(value: number, currencyCode = "€", separators: {decimal?: string, thousand?: string} = {}) {
+    // First obtain formatted value with '.' as decimal separator and ',' as thousand separator (US notation).
+    let formattedValue = (Math.ceil(value) / 100).toLocaleString("en-US", {minimumFractionDigits: 2, maximumFractionDigits: 2});
```
Comment (dcbr) [inline src/formatting.ts:3]: I kept the `Math.ceil` here, but I was wondering if this makes sense? You can't really pay a 0.1 c€ right? So I wondered if this shouldn't be replaced by `Math.round` or even `Math.floor`.

**Comment by samjmck** [comment]
Initial impressions look very good! I have exams coming up soon but will try my best to properly review the PR and merge ASAP.

**Comment by dcbr** [comment]
No hurry, take your time and good luck with the exams!
The only thing I was wondering at the moment is why the tests failed; as I ran them locally (in a github codespace) and there everything worked fine. So I'm not sure what's causing the failure here.

**Comment by samjmck** [comment]
The downloading of deno dependencies sometimes throws an error for some reason.
```
error: TS2498 [ERROR]: Module '"deno:///missing_dependency.d.ts"' uses 'export =' and cannot be used with 'export *'.
export * from "./objects.d.ts";
              ~~~~~~~~~~~~~~~~
    at https://cdn.skypack.dev/-/pdf-lib@v1.17.1-P8YmSI9gov9vMnrV7v2f/dist=es2019,mode=types/cjs/api/index.d.ts:6:15
Error: Process completed with exit code 1.
```

This is despite locking the dependencies to a fixed version. I've rerun the tests and it's all fine.

File: src/pdf.ts [hunk 2]
```diff
@@ -68,44 +109,115 @@ export async function fillPdf(
...
+    setFieldText(totalTaxValue, formatMoney(params.totalTaxValue, moneyCur, moneySep), font);
+    setFieldText(locationField, params.location, font);
+    setFieldText(dateField, params.date, font);
 
     if(params.signaturePng) {
         signatureField.setImage(await document.embedPng(params.signaturePng));
     }
-    signatureNameField.setText(params.signatureName);
-    signatureCapacityField.setText(params.signatureCapacity);
+    setFieldText(signerField, `${params.signatureName}  , ${params.signatureCapacity}`, font);
```
Comment (dcbr) [inline src/pdf.ts:147]: Probably best to omit the comma if no capacity is provided.
```suggestion
    let signer = params.signatureName;
    if (params.signatureCapacity.length > 0) {
        signer += `  , ${params.signatureCapacity}`;
    }
    setFieldText(signerField, signer, font);
```

**Comment by samjmck** [comment]
Amazing work, I really appreciate the time and effort you put into this. I received an email a few months ago asking to add different languages to the PDFs, so I'm sure many users will be happy with this PR as well. The PDF generation code seems like it would've been a real hassle to learn and write, I remember uploading the base PDF to a site to add input boxes manually. Great job!

**Comment by dcbr** [comment]
Thanks, I hope it will be useful indeed :) And to be honest, I wouldn't have been able to write it entirely from scratch either. I had some experience with PDF text modifications in python already, so that helped a lot. And the `pdf-lib` library also does a great job to simplify a lot of things. The whole PDF specification is actually quite insane really, it keeps on surprising me whenever I learn something new about it (mostly in a bad way 😆). But it's always nice if you can get it to work properly at the end of the day :)

**Pull request merged**